### PR TITLE
Add common configuration patterns to documentation

### PR DESCRIPTION
**Pull request by zigaLuksic**
I know i made a lot of typos, you know i made a lot of typos.
Let's avoid the back and forth in the review, just push a commit to the appropriate branch. You can voice any other concerns here or ping me directly.

Relevant [docs](https://eo-grow.readthedocs.io/en/add-tips-and-tricks-docs/common-configuration-patterns.html)

We should add tips and tricks for working in Jupyter as well (separate MR to limit the scope).

### CODE REVIEW

File: docs/source/common-configuration-patterns.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    "capture_warnings": false
+  },
+  "feature": ["data", "BANDS"],
+  "map_dtype": "int16",
+  "cogify": true,
+  ...
+}
+```
+
+The processed configuration will have all the logging settings from `global_config.json`, except for `"capture_warnings"`. See config language rules for config joins.
```
Comment (mlubej) [inline docs/source/common-configuration-patterns.md:61]: Is there a link for this?

Comment (zigaLuksic) [reply docs/source/common-configuration-patterns.md:61]: we'd need to check how to link to other MD-as-RST documents. Perhaps that sentence can just go out?

Comment (mlubej) [reply docs/source/common-configuration-patterns.md:61]: Is there a link for this?

Comment (mlubej) [reply docs/source/common-configuration-patterns.md]: managed to find the magic combo, link works now.

Comment (zigaLuksic) [reply docs/source/common-configuration-patterns.md]: cool, will merge on test success unless you have additional plans

**Review by mlubej — APPROVED** [review]

**Pull request merged**